### ISSUE TITLE
Setting global marker throws "void-variable: alist" error 

### ISSUE THREAD

**Issue by hackeryarn** [hlissner/doom-emacs#2570]
**What did you expect to happen?**

When I press `m M` in normal mode or try to set any other global mark (capital letter mark), I expect emacs to set the mark.

**What actually happened?**

When I execute `m M` in normal mode, I get the error `let: Symbol’s value as variable is void: alist` and emacs does not set a global mark.

**Additional details:**

I have not made any modifications to evil keybinding and run a mostly unmodified doom-emacs instance.

**Error trace**:

<details>

```
Debugger entered--Lisp error: (void-variable alist)
  symbol-value(alist)
  (let ((tail (symbol-value list-var))) (while (and tail (not (equal (car-safe (car-safe tail)) key))) (setq tail (cdr tail))) (if tail (setcar tail (cons key val)) (set list-var (append (symbol-value list-var) (list (cons key val))))) (if elements (apply (function evil-add-to-alist) list-var elements) (symbol-value list-var)))
  evil-add-to-alist(alist 77 #<marker in no buffer>)
  (cond ((and marker (symbolp marker) (boundp marker)) (set marker (or (symbol-value marker) (make-marker))) (setq marker (symbol-value marker))) ((eq marker (quote evil-jump-backward-swap)) (evil-set-jump) (throw (quote done) nil)) ((functionp marker) (user-error "Cannot set special marker `%c'" char)) ((evil-global-marker-p char) (setq alist (default-value (quote evil-markers-alist)) marker (make-marker)) (evil-add-to-alist (quote alist) char marker) (setq-default evil-markers-alist alist)) (t (setq marker (make-marker)) (evil-add-to-alist (quote evil-markers-alist) char marker)))
  (if (markerp marker) nil (cond ((and marker (symbolp marker) (boundp marker)) (set marker (or (symbol-value marker) (make-marker))) (setq marker (symbol-value marker))) ((eq marker (quote evil-jump-backward-swap)) (evil-set-jump) (throw (quote done) nil)) ((functionp marker) (user-error "Cannot set special marker `%c'" char)) ((evil-global-marker-p char) (setq alist (default-value (quote evil-markers-alist)) marker (make-marker)) (evil-add-to-alist (quote alist) char marker) (setq-default evil-markers-alist alist)) (t (setq marker (make-marker)) (evil-add-to-alist (quote evil-markers-alist) char marker))))
  (let ((marker (evil-get-marker char t)) alist) (if (markerp marker) nil (cond ((and marker (symbolp marker) (boundp marker)) (set marker (or (symbol-value marker) (make-marker))) (setq marker (symbol-value marker))) ((eq marker (quote evil-jump-backward-swap)) (evil-set-jump) (throw (quote done) nil)) ((functionp marker) (user-error "Cannot set special marker `%c'" char)) ((evil-global-marker-p char) (setq alist (default-value (quote evil-markers-alist)) marker (make-marker)) (evil-add-to-alist (quote alist) char marker) (setq-default evil-markers-alist alist)) (t (setq marker (make-marker)) (evil-add-to-alist (quote evil-markers-alist) char marker)))) (add-hook (quote kill-buffer-hook) (function evil-swap-out-markers) nil t) (set-marker-insertion-type marker advance) (set-marker marker (or pos (point))))
  (catch (quote done) (let ((marker (evil-get-marker char t)) alist) (if (markerp marker) nil (cond ((and marker (symbolp marker) (boundp marker)) (set marker (or (symbol-value marker) (make-marker))) (setq marker (symbol-value marker))) ((eq marker (quote evil-jump-backward-swap)) (evil-set-jump) (throw (quote done) nil)) ((functionp marker) (user-error "Cannot set special marker `%c'" char)) ((evil-global-marker-p char) (setq alist (default-value (quote evil-markers-alist)) marker (make-marker)) (evil-add-to-alist (quote alist) char marker) (setq-default evil-markers-alist alist)) (t (setq marker (make-marker)) (evil-add-to-alist (quote evil-markers-alist) char marker)))) (add-hook (quote kill-buffer-hook) (function evil-swap-out-markers) nil t) (set-marker-insertion-type marker advance) (set-marker marker (or pos (point)))))
  (closure (t) (char &optional pos advance) "Set the marker denoted by CHAR to position POS.\nPOS defaults to the current position of point.\nIf ADVANCE is t, the marker advances when inserting text at it;\notherwise, it stays behind." (interactive (list (read-char))) (catch (quote done) (let ((marker (evil-get-marker char t)) alist) (if (markerp marker) nil (cond ((and marker (symbolp marker) (boundp marker)) (set marker (or ... ...)) (setq marker (symbol-value marker))) ((eq marker (quote evil-jump-backward-swap)) (evil-set-jump) (throw (quote done) nil)) ((functionp marker) (user-error "Cannot set special marker `%c'" char)) ((evil-global-marker-p char) (setq alist (default-value ...) marker (make-marker)) (evil-add-to-alist (quote alist) char marker) (setq-default evil-markers-alist alist)) (t (setq marker (make-marker)) (evil-add-to-alist (quote evil-markers-alist) char marker)))) (add-hook (quote kill-buffer-hook) (function evil-swap-out-markers) nil t) (set-marker-insertion-type marker advance) (set-marker marker (or pos (point))))))(77)
  apply((closure (t) (char &optional pos advance) "Set the marker denoted by CHAR to position POS.\nPOS defaults to the current position of point.\nIf ADVANCE is t, the marker advances when inserting text at it;\notherwise, it stays behind." (interactive (list (read-char))) (catch (quote done) (let ((marker (evil-get-marker char t)) alist) (if (markerp marker) nil (cond ((and marker ... ...) (set marker ...) (setq marker ...)) ((eq marker ...) (evil-set-jump) (throw ... nil)) ((functionp marker) (user-error "Cannot set special marker `%c'" char)) ((evil-global-marker-p char) (setq alist ... marker ...) (evil-add-to-alist ... char marker) (setq-default evil-markers-alist alist)) (t (setq marker ...) (evil-add-to-alist ... char marker)))) (add-hook (quote kill-buffer-hook) (function evil-swap-out-markers) nil t) (set-marker-insertion-type marker advance) (set-marker marker (or pos (point)))))) 77)
  evil-set-marker(77)
  funcall-interactively(evil-set-marker 77)
  call-interactively(evil-set-marker nil nil)
  command-execute(evil-set-marker)
```
</details>

**System information:**

<details>

```
emacs   version    26.3
        features   XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GSETTINGS GLIB NOTIFY ACL GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS LIBSYSTEMD LCMS2
        build      Aug 29, 2019
        buildopts  (--prefix=/usr --sysconfdir=/etc --libexecdir=/usr/lib --localstatedir=/var --with-x-toolkit=gtk3 --with-xft --with-modules 'CFLAGS=-march=x86-64 -mtune=generic -O2 -pipe -fno-plt' CPPFLAGS=-D_FORTIFY_SOURCE=2 LDFLAGS=-Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now)
        windowsys  x
        daemonp    server-running
doom    version    2.0.9
        build      HEAD -> develop 21db1b7c9 2020-02-13 14:45:08 -0500
        dir        ~/.doom.d/
system  type       gnu/linux
        config     x86_64-pc-linux-gnu
        shell      /bin/bash
        uname      Linux 5.4.18-1-MANJARO #1 SMP PREEMPT Thu Feb 6 11:41:30 UTC 2020 x86_64
        path       (~/.nvm/versions/node/v8.15.0/bin ~/.pyenv/shims ~/.pyenv/plugins/pyenv-virtualenv/shims ~/.local/bin /usr/local/sbin /usr/local/bin /usr/bin /usr/lib/jvm/default/bin /usr/bin/site_perl /usr/bin/vendor_perl /usr/bin/core_perl /var/lib/snapd/snap/bin /usr/lib/emacs/26.3/x86_64-pc-linux-gnu/)
config  envfile    envvar-file
        elc-files  0
        modules    (:completion company ivy :ui deft doom doom-dashboard doom-quit hl-todo hydra modeline nav-flash ophints (popup +all +defaults) vc-gutter vi-tilde-fringe window-select workspaces :editor (evil +everywhere) file-templates fold multiple-cursors parinfer rotate-text snippets :emacs dired electric ibuffer vc :term eshell :checkers syntax spell grammar :tools ansible docker (eval +overlay) (lookup +docsets) magit make pdf rgb :lang clojure common-lisp data elm emacs-lisp go (haskell +dante) hy javascript markdown (org +journal +dragndrop +present) plantuml (python +pyenv) racket rest rust sh web :email (mu4e +gmail) :app irc :config (default +bindings +smartparens) :private org)
        packages   (n/a)
        elpa       (n/a)
        unpin      (n/a)
```
</details>

**Comment by hackeryarn** [comment]
Sorry, I didn't dig deep enough into the evil issue log. 

This is a `evil-mode` issue https://github.com/emacs-evil/evil/issues/1270.

**Comment by hlissner** [comment]
i'll keep this open until it is resolved upstream; because I'll still need to bump the pinned version of evil once it has.

**Comment by TheBB** [comment]
Should be fixed in evil as of 810eaf6.
